common/middleware/date-utils.js

```diff
@@ -0,0 +1,22 @@
+const {
+  getDateFromParams,
+  getDateRange,
+} = require('../../common/helpers/date-utils')
+
+function setPeriod(req, res, next, period) {
```

Comment:
I'd be tempted to split these out actually. I've been looking at the middleware file in moves and wanted to split them out into separate files as I felt like it was getting too large to manage.

The pattern is normally to just use a `middleware.js` file for 1 or 2 middlewares, then if it grows, split into separate files but include a `middleware/index.js` loader file. That way, any previous imports aren't actually broken and it allows for expansion.

Reply:
So here I'd maybe create `./common/middleware/set-date-period.js` and `./common/middleware/set-date-range.js`. Makes them nice and atomic and testable.

Reply:
Yes, makes sense. I'll do it.